lib/modules/datasource/aws-eks-addon/index.spec.ts

```diff
@@ -0,0 +1,153 @@
+import {
+  type AddonInfo,
+  type AddonVersionInfo,
+  DescribeAddonVersionsCommand,
+  DescribeAddonVersionsResponse,
+  EKSClient,
+} from '@aws-sdk/client-eks';
+import { mockClient } from 'aws-sdk-client-mock';
+import { getPkgReleases } from '..';
+import { AwsEKSAddonDataSource } from '.';
+
+const datasource = AwsEKSAddonDataSource.id;
+const eksMock = mockClient(EKSClient);
+
+const addonVersion1: AddonVersionInfo = {
+  addonVersion: 'v1.18.1-eksbuild.1',
+  architecture: ['amd64', 'arm64'],
+  compatibilities: [
+    {
+      clusterVersion: '1.30',
+      platformVersions: ['*'],
+      defaultVersion: false,
+    },
+  ],
+  requiresConfiguration: false,
+};
+
+const addonVersion2: AddonVersionInfo = {
+  addonVersion: 'v1.18.2-eksbuild.1',
+  architecture: ['amd64', 'arm64'],
+  compatibilities: [
+    {
+      clusterVersion: '1.30',
+      platformVersions: ['*'],
+      defaultVersion: false,
+    },
+  ],
+  requiresConfiguration: false,
+};
+
+// a bad addonVersion that's missing the basic fields.
+const addonVersionBad: AddonVersionInfo = {};
+
+/**
+ * Testdata for mock implementation of EKSClient
+ */
+const vpcCniAddonInfo: AddonInfo = {
+  addonName: 'vpc-cni',
+  type: 'networking',
+  addonVersions: [addonVersion1, addonVersion2, addonVersionBad],
+  publisher: 'eks',
+  owner: 'aws',
+};
+
+const mockAddon: DescribeAddonVersionsResponse = {
+  addons: [vpcCniAddonInfo],
+};
+
+const mockEmpty: DescribeAddonVersionsResponse = {
+  addons: [],
+};
+
+const mockNull: DescribeAddonVersionsResponse = {};
+
+const mockNullAddonVersions: DescribeAddonVersionsResponse = {
+  addons: [
+    {
+      addonName: 'non-existing-addon',
+      type: 'networking',
+      publisher: 'eks',
+      owner: 'aws',
+      // missing addonVersions
+    },
+  ],
+};
+
+function mockDescribeAddonVersionsCommand(
+  result: DescribeAddonVersionsResponse,
+): void {
+  eksMock.reset();
+  eksMock.on(DescribeAddonVersionsCommand).resolves(result);
+}
+
+function mockDescribeAddonVersionsCommandWithRegion(
+  result: DescribeAddonVersionsResponse,
+): void {
+  eksMock.reset();
+  eksMock
+    .on(DescribeAddonVersionsCommand)
+    .callsFake(async (input, getClient) => {
+      const client = getClient();
+      const region = await client.config.region();
+      return {
+        ...result,
+        // put the client region as nextToken
+        // so that when we assert on the snapshot, we also verify that region from packageName is
+        // passed to aws client.
+        nextToken: region,
+      };
+    });
+}
+
+describe('modules/datasource/aws-eks-addon/index', () => {
+  describe('getPkgReleases()', () => {
+    it.each([
```

Comment:
please use string template syntax

Reply:
Updated, I hope I'm doing what you have in mind, please take a look